src/main/java/Duke.java

```diff
@@ -1,10 +1,308 @@
+            + LS + "in the format " + MESSAGE_COMMAND_EVENT_FORMAT + ", where X is the event and Y is the date!";
+
+    /** List of all tasks (Event, Deadline, Todo all inherit 'Task' class) */
+    private static Task[] tasks = new Task[100];
+
+    /**
+     * Prints lines of messages. Can take in variable number of arguments.
+     *
+     * @param lines Strings to be printed, each on a new line
+     */
+    public static void showMessage(String... lines) {
+        for (String line : lines) {
+            System.out.println(line);
+        }
+    }
+
+    /**
+     * Prints lines of messages framed by divider. Can take in variable number of arguments.
+     *
+     * @param lines Strings to be printed, each on a new line
+     */
+    public static void showMessageFramedWithDivider(String... lines) {
+        System.out.println(DIVIDER);
+        for (String line : lines) {
+            System.out.println(line);
+        }
+        System.out.println(DIVIDER);
+    }
+
+    /**
+     * Prints list of commands
+     */
+    public static void showListOfCommands() {
+        showMessageFramedWithDivider(MESSAGE_COMMAND_LIST);
+    }
+
+    /**
+     * Prints Welcome message and list of commands
+     */
+    public static void welcome() {
+        showMessageFramedWithDivider(MESSAGE_WELCOME_DUDE);
+        showListOfCommands();
+    }
+
+    /**
+     * Prints Goodbye message and exits the program.
+     */
+    public static void exit() {
+        showMessageFramedWithDivider(MESSAGE_BYE);
+        System.exit(0);
+    }
+
+    /**
+     * Prints current tasks
+     */
+    public static void printTasks() {
+        if (Task.getNumTasks() == 0) {
+            showMessageFramedWithDivider(MESSAGE_NO_TASKS_YET);
+        } else {
+            showMessage(DIVIDER, MESSAGE_INTRODUCE_TASKS);
+            for (int i = 1; i <= Task.getNumTasks(); i++) {
+                System.out.println(i + ". " + tasks[i]);
+            }
+            showMessage(DIVIDER);
+        }
+    }
+
+    /**
+     * Returns a String array where 0th index is command string and 1st index is the remaining parameters
+     * Command string and parameter string is assumed to be separated by the first " " in input
+     * If no parameters are provided in the input, 1st index will be set to EMPTY
+     *
+     * @param input Raw user input string
+     * @return String array [command, parameters]
+     */
+    public static String[] splitInputIntoCommandAndParams(String input) {
+        String[] commandAndParams = new String[2];
+        final String[] splitInput = input.trim().split(" ", 2);
+        //command string
+        commandAndParams[0] = splitInput[0];
+        //param string, if not given, set to EMPTY for error handling
+        commandAndParams[1] = (splitInput.length >= 2) ? splitInput[1] : EMPTY;
+        return commandAndParams;
+    }
+
+    /**
```

Comment:
It is impressive that you have added comments to explain your codes clearly to the reviewers. Thanks!! :)))